include/tnt/core/impl/tensor_impl.hpp

```diff
@@ -41,6 +57,16 @@ inline Tensor<DataType>::Tensor(const Shape& shape, const PtrType& data)
     this->data   = data;
 }
 
+TEST_CASE_TEMPLATE("Tensor(Shape&, PtrType&)", T, test_data_types)
+{
+    Shape s1;
+    AlignedPtr<T> ptr1;
```

Comment:
This is an empty pointer and empty shape. The test would be better if you set the shape and filled the pointer with some values. For example- 
```
Shape s1 = Shape{2, 2};
AlignedPtr<T> ptr1 = {1, 2, 3, 4};
```

Then you can explicitly test if tensor shape == {2, 2} and aligned ptr data = {1, 2, 3, 4}